server/jetstream_errors.go

```diff
@@ -84,100 +84,115 @@ const (
+		JSStreamInvalidConfigF:             {Code: 500, ErrCode: "JS052", Description: "%s"},
+		JSStreamLimitsErrF:                 {Code: 500, ErrCode: "JS053", Description: "%s"},
+		JSStreamMessageExceedsMaximumErr:   {Code: 400, ErrCode: "JS054", Description: "message size exceeds maximum allowed"},
+		JSStreamMirrorNotUpdatableErr:      {Code: 400, ErrCode: "JS055", Description: "Mirror configuration can not be updated"},
+		JSStreamMismatchErr:                {Code: 400, ErrCode: "JS056", Description: "stream name in subject does not match request"},
+		JSStreamMsgDeleteFailedF:           {Code: 500, ErrCode: "JS057", Description: "%s"},
+		JSStreamNameExistErr:               {Code: 400, ErrCode: "JS058", Description: "stream name already in use"},
+		JSStreamNotFoundErr:                {Code: 404, ErrCode: "JS059", Description: "stream not found"},
+		JSStreamNotMatchErr:                {Code: 400, ErrCode: "JS060", Description: "expected stream does not match"},
+		JSStreamReplicasNotUpdatableErr:    {Code: 400, ErrCode: "JS061", Description: "Replicas configuration can not be updated"},
+		JSStreamRestoreErrF:                {Code: 500, ErrCode: "JS062", Description: "restore failed: %s"},
+		JSStreamSequenceNotMatchErr:        {Code: 503, ErrCode: "JS063", Description: "expected stream sequence does not match"},
+		JSStreamSnapshotErrF:               {Code: 500, ErrCode: "JS064", Description: "snapshot failed: %s"},
+		JSStreamSubjectOverlapErr:          {Code: 500, ErrCode: "JS065", Description: "subjects overlap with an existing stream"},
+		JSStreamTemplateCreateErrF:         {Code: 500, ErrCode: "JS066", Description: "%s"},
+		JSStreamTemplateDeleteErrF:         {Code: 500, ErrCode: "JS067", Description: "%s"},
+		JSStreamTemplateNotFoundErr:        {Code: 404, ErrCode: "JS068", Description: "template not found"},
+		JSStreamUpdateErrF:                 {Code: 500, ErrCode: "JS069", Description: "%s"},
+		JSStreamWrongLastMsgIDErrF:         {Code: 400, ErrCode: "JS070", Description: "wrong last msg ID: %s"},
+		JSStreamWrongLastSequenceErrF:      {Code: 400, ErrCode: "JS071", Description: "wrong last sequence: %d"},
+		JSTempStorageFailedErr:             {Code: 500, ErrCode: "JS072", Description: "JetStream unable to open temp storage for restore"},
+		JSTemplateNameNotMatchSubjectErr:   {Code: 400, ErrCode: "JS073", Description: "template name in subject does not match request"},
 	}
 )
 
+type natsCodedError interface {
+	natsErrCode() string
+}
+
+func IsNatsCodedErr(err error, code string) bool {
+	ce, ok := err.(natsCodedError)
+	if !ok {
+		return false
+	}
+
+	return ce.natsErrCode() == code
+}
+
 // ApiError is included in all responses if there was an error.
 // TODO(dlc) - Move to more generic location.
 type ApiError struct {
 	Code        int    `json:"code"`
-	Kind        string `json:"kind,omitempty"`
+	ErrCode     string `json:"err_code,omitempty"`
```

Comment:
I think this should be u16.

Reply:
agree,

Not sure why I cant comment directly by the:

> We should leave, no harm imo..

I think there is plenty of harm tbh, in order to return the code we have to use a different type - we wont use this constant anymore so it'll just cause confusion.

We could as I say do `ErrJetStreamNotEnabled = ApiErrors[xxx]` but that only works for the kind of error without any dynamic content so would work in this specific case but in general not.

Reply:
ok maybe you are correct and since we made the functions private no one could really run something and use those.